mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs

```diff
@@ -1029,9 +1029,8 @@ impl<MakerCoin: MmCoin + MakerCoinSwapOpsV2, TakerCoin: MmCoin + TakerCoinSwapOp
             },
         };
 
-        let total_payment_value = &(&state_machine.taker_volume + &state_machine.dex_fee().total_spend_amount())
-            + &state_machine.taker_premium;
-        let preimage_value = TradePreimageValue::Exact(total_payment_value.to_decimal());
+        let payment_value_with_premium = &state_machine.taker_volume + &state_machine.taker_premium;
+        let preimage_value = TradePreimageValue::Exact(payment_value_with_premium.to_decimal());
         let stage = FeeApproxStage::StartSwap;
 
         let taker_payment_fee = match state_machine
```

Comment:
i guess we gotta add the dex fee to the argument of `get_sender_trade_fee` for correctness.

Reply:
In fact we should refactor this:
I think, for TPU we should not use this get_sender_trade_fee fn but implement new fee estimation functions corresponding to the TPU steps (send funding and spend funding). 
I though we could do this in a new PR and in this PR just do a quick fix to allow to spend max amount (which actually is needed to run swaps with LR)